Code/Tools/LuaIDE/Source/LUA/WatchesPanel.cpp

```diff
@@ -831,7 +831,7 @@ bool WatchesDataModel::setData (const QModelIndex& index, const QVariant& value,
 
 const char* WatchesDataModel::SafetyType(char c) const
 {
-    if (c <= LUA_TNONE || c > LUA_NUMTAGS)
+    if (static_cast<int>(c) <= LUA_TNONE || c > LUA_NUMTAGS)
```

Comment:
Doesn't the greater than comparison also need to cast the character to an int.
```suggestion
    if (static_cast<int>(c) <= LUA_TNONE || static_cast<int>(c) > LUA_NUMTAGS)
```

Reply:
`LUA_TNONE` was defined as `-1`, so the compiler complained because apparently `char` comparisons to a negative is a warning -> error

Reply:
Ok, so LUA_NUMTAGS is positive `9` and therefore that comparison doesn't need a cast then.

Thanks for the clarification.